index.js

```diff
@@ -31,37 +31,37 @@
-//:                               \         / \         / \
-//:                                \       /   \       /   \
-//:                                 \     /     \     /     \
-//:                                  \   /       \   /       \
-//:                                   \ /         \ /         \
-//:                               Alternative    Monad     ChainRec
-//:                                                       (chainRec)
+//:  Setoid   Semigroupoid   Semigroup   Foldable        Functor      Contravariant
+//: (equals)   (compose)     (concat)    (reduce)         (map)        (contramap)
+//:     |          |             |           \         / | | | | \
+//:     |          |             |            \       /  | | | |  \
+//:     |          |             |             \     /   | | | |   \
+//:     |          |             |              \   /    | | | |    \
+//:     |          |             |               \ /     | | | |     \
+//:    Ord      Category       Monoid        Traversable | | | |      \
+//:   (lte)       (id)         (empty)       (traverse)  / | | \       \
+//:                                                     /  | |  \       \
+//:                                                    /   / \   \       \
+//:                                            Profunctor /   \ Bifunctor \
+//:                                             (promap) /     \ (bimap)   \
+//:                                                     /       \           \
+//:                                                    /         \           \
+//:                                                  Alt        Apply      Extend
+//:                                                 (alt)        (ap)     (extend)
+//:                                                 /            /  \           \
+//:                                                /            /    \           \
+//:                                               /            /      \           \
+//:                                              /            /        \           \
+//:                                             /            /          \           \
+//:                                           Plus     Applicative     Chain      Comonad
+//:                                          (zero)       (of)        (chain)     (extract)
+//:                                             \         /  \         /  \
+//:                                              \       /    \       /    \
+//:                                               \     /      \     /      \
+//:                                                \   /        \   /        \
+//:                                                 \ /          \ /          \
+//:                                              Alternative    Monad      ChainRec
+//:                                                                       (chainRec)
```

Comment:
I was attempting to fix the alignment of branches. I might leave the fiddling to you though David if that's okay.